### ISSUE TITLE
Add Support for Azure Frontdoor

### ISSUE THREAD

**Issue by Theragus** [infracost/infracost#909]
Terraform Resource: 

`azurerm_frontdoor`

Pricing Information:

https://azure.microsoft.com/en-us/pricing/details/frontdoor/

**Comment by aliscott** [comment]
Thanks @Theragus!

Adding some details for the cost components, cc @alikhajeh1 and @tim775 for the open question.

### Open questions:

1. Frontend hosts are charged for when there are >100. We can count the number by counting the `frontend_endpoint` attributes of the `azurerm_frontdoor` resource, but I can't work out if the 100 limit is per-account or per-frontdoor instance. Options are:
  a. Ignore this for now
  b. Always show price for the frontend hosts but have a `-` cost
  c. Assume per-frontdoor instance and only show the cost for these if there are >100.

### azurerm_frontdoor

Terraform resource: https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/frontdoor

Cost components:

* Outbound data transfer - all of these should have unit `GB`
  * North America, Europe (first 10TB)
  * North America, Europe (next 40TB)
  * North America, Europe (next 100TB)
  * Asia Pacific (including Japan) (first 10TB)
  * Asia Pacific (including Japan) (next 40TB)
  * Asia Pacific (including Japan) (next 100TB)
  * South America (first 10TB)
  * South America (next 40TB)
  * South America (next 100TB)
  * Australia (first 10TB)
  * Australia (next 40TB)
  * Australia (next 100TB)
  * India (first 10TB)
  * India (next 40TB)
  * India (next 100TB)
* Inbound data transfer - unit `GB`
* Routing rules (first 5 rules) - unit `rules`
* Routing rules (per additional rule) - unit `rules`

Notes:

* For "Outbound data transfer", if no usage is passed in we should only show the prices for the "X (first 10TB)" cost components. If usage is passed in we should only show the relevant ones. This is similar to what we do for the [aws_cloudfront_distribution](https://github.com/infracost/infracost/blob/master/internal/providers/terraform/aws/cloudfront_distribution.go) resource.
* For "Routing rules" we should only show the "(per additional rule)" cost component if there are more than 5 rules. We can find the number of routing rules by finding the number of `routing_rule` attributes on the resource.

Usage file:

```yml
  azurerm_frontdoor.my_frontdoor:
    monthly_outbound_data_transfer_gb: # Monthly outbound data transfer from the following, in GB:
      north_america_europe: 51200000 # North America, Europe
      asia_pacific: 220000           # Asia Pacific (including Japan)
      south_america: 10000           # South America
      australia: 50000               # Australia
      india: 387000                  # India
    monthly_inbound_data_transfer_gb: 1000 # Monthly inbound data transfer in GB
```

### azurerm_frontdoor_firewall_policy

Terraform resource: https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/frontdoor_firewall_policy

Cost components:

* Policy - unit `months`
* Custom rules - unit `rules`
* Custom rule requests - unit `1M requests`
* Managed rulesets - unit `rulesets`
* Managed ruleset requests - unit `1M requests`

Notes:

* For "Custom rules" the quantity can be retrieved by counting the number of `custom_rule` attributes.
* For "Managed rulesets" the quantity can be retrieved by counting the number of `managed_rule` attributes.

Usage file:

```yml
  azurerm_frontdoor_firewall_policy.my_frontdoor_firewall_policy:
    monthly_custom_rule_requests: 10000 # Monthly number of custom rule requests
    monthly_managed_ruleset_requests: 10000 # Monthly number of managed ruleset requests
```

### Free resources

* azurerm_frontdoor_custom_https_configuration

### Unsupported resources

Azure Front Door Standard/Premium (this isn't supported by Terraform yet: https://github.com/hashicorp/terraform-provider-azurerm/issues/11983)

**Comment by alikhajeh1** [comment]
@aliscott should we only show "North America, Europe (first 10TB)" by default unless usage-file params exist for the other ones? That's what we ended-up doing for CloudFront to simplify the output

**Comment by aliscott** [comment]
> @aliscott should we only show "North America, Europe (first 10TB)" by default unless usage-file params exist for the other ones? That's what we ended-up doing for CloudFront to simplify the output

+1 good point. This is now how we are doing it for other resources.

**Comment by alikhajeh1** [comment]
Thanks @aliscott - I updated the desc, I also tweaked it with my thinking for the open question.

**Comment by vdmgolub** [comment]
This has been merged to master and will go live with the v0.9.14 release sometime next week.

**Comment by tim775** [comment]
@Theragus this is live in v0.9.14
